tests/core/full_node/test_mempool.py

```diff
@@ -159,6 +165,54 @@ async def test_basic_mempool_manager(self, two_nodes):
             spend_bundle.name(),
         )
 
+    # this test makes sure that one spend successfully asserts the announce from
+    # another spend, even though the assert condition is duplicated 100 times
+    @pytest.mark.asyncio
```

Comment:
This is testing asserting condition from the same bundle right? it's being aggregated `SpendBundle.aggregate([spend_bundle1, spend_bundle2])`

Reply:
Asserting announcements from a different mempool item was not allowed on purpose.

Reply:
right, I use the term "spend" to refer to a coin that's being spent. it's all in the same spend bundle. I don't know of a better name

Reply:
this test makes sure that announcements aren't *just* matched within a spent coin, but at the whole spend bundle/block level. The way we parse and store conditions today makes this quite obvious, but in the rust implementation some conditions are validated locally, within a spend